src/services/addressResolver/index.test.ts

```diff
@@ -0,0 +1,72 @@
+import axios from "axios";
+import { getIdentityName } from "./index";
+import { ThirdPartyAPIEntryProps } from "./../../common/hooks/useThirdPartyAPIEndpoints";
+
+jest.mock("axios");
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+it("should return the first name if it can be found with any resolver", async () => {
+  mockedAxios.get.mockResolvedValueOnce({
+    data: {
+      identity: {
+        identifier: "0xA",
+        name: "ABC Pte Ltd",
+        remarks: "Added by Raymond",
+      },
+    },
+  });
+
+  const endpoints = [
+    {
+      id: "1",
+      name: "demo",
+      endpoint: "https://demo.io/identifier/",
+    },
+  ];
+
+  const resolveIdentityName = async () => {
+    const identityName = await getIdentityName(endpoints, "0xA");
+    expect(identityName).toEqual("ABC Pte Ltd");
+  };
+
+  resolveIdentityName();
+});
+
+it("should return undefined if it cannot be resolved anywhere", async () => {
+  mockedAxios.get.mockResolvedValueOnce({
```

Comment:
Does this resolve or get rejected? The server implementation is suppose to return 404, it should cause a rejection

Reply:
um actually i don't really understand and am unsure